src/librbd/api/Trash.h

```diff
@@ -22,11 +22,16 @@ template <typename ImageCtxT = librbd::ImageCtx>
 struct Trash {
 
   static int move(librados::IoCtx &io_ctx, rbd_trash_image_source_t source,
-                  const std::string &image_name, uint64_t delay);
+                  const std::string &image_name, uint64_t delay,
+                  bool check_for_watchers);
+  static int move(librados::IoCtx &io_ctx, rbd_trash_image_source_t source,
+                  const std::string &image_name, const std::string &image_id,
+                  uint64_t delay, bool check_for_watchers);
   static int get(librados::IoCtx &io_ctx, const std::string &id,
                  trash_image_info_t *info);
   static int list(librados::IoCtx &io_ctx,
-                  std::vector<trash_image_info_t> &entries);
+                  std::vector<trash_image_info_t> &entries,
+                  bool exclude_user_remove_source);
```

Comment:
@dillaman Actually when you were talking about excluding "removing" images from trash list output, I thought you meant `rbd trash list` without `--all` flag (sorry I missed you were talking about API). I would still like to see those images in `rbd trash list --all` output. But if you still prefer this way, no objections from me.

Reply:
I just wanted to avoid double-listing the image and keep this "move-to-trash-on-remove" logic as an internal implementation detail of librbd. I think if I can clean up some of the failure mode rough-edges, it should be fine.